centaur/src/main/scala/centaur/CromwellDatabase.scala

```diff
@@ -10,10 +10,13 @@ trait CromwellDatabase {
 }
 
 object CromwellDatabase {
+
+  lazy val instance: CromwellDatabase = fromConfig(CentaurConfig.conf)
```

Comment:
I'm not sure what effect it might have to change this object's lifecycle. Maybe worth a quick chat.

Reply:
Discussed IRL. The `object ErrorReporters` already effectively had its own singleton of the database object. So no change is expected around `ErrorReporters`, which was not closing the database `instance`.

There is a line in this PR that also used the same `instance` for tests extending from `AbstractCromwellEngineOrBackendUpgradeTestCaseSpec`. `
From the discussion, there's no reason that each `AbstractCromwellEngineOrBackendUpgradeTestCaseSpec` needs its own connection pool(s... ackchyually one for engine, one for metadata). AbstractCromwellEngineOrBackendUpgradeTestCaseSpec` currently does not close its instance. The `AbstractCromwellEngineOrBackendUpgradeTestCaseSpec` test \_do\_ drop and re-create the database \_schemas\_. However, the tests can do that existing work through connections retrieved from the singleton connection pool `instance` versus creating their own connection pool(s).

In CI testing all the tests meet previous regression expectations. If we decide to not use a singleton connection pool(s) in the future for any reason we can switch `AbstractCromwellEngineOrBackendUpgradeTestCaseSpec` back to creating its own separate connections.